test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test "creates valid reply" do
+    login_as(@admin)
+    post :reply, params: { message: { subject: "Valid Message", :body => "message with subject" }, id: @message.id }
+
+    MessageWorker.drain
+    msg = Message.last
+    assert "sent", msg.state
+    assert_equal "message with subject", msg.body
```

Comment:
Instead of only going to the DB to fetch a message and test it out, we could also have assertions that depend on the code that's called in the controller.
Things like
```ruby
assert_equal flash[:notice], 'Reply was sent.'
assert_equal response.redirect_url, 'some url we're expecting...'
```
and so on.

Reply:
Updated assertions